From f4444a81218ede5eb58306bd57eaefb5d9ffd9e2 Mon Sep 17 00:00:00 2001
From: Fabian Meumertzheim <fabian@meumertzhe.im>
Date: Mon, 8 Jan 2024 13:30:32 +0100
Subject: [PATCH] Disable unsupported `layering_check` Bazel feature

This allows downstream projects to use `layering_check` without having
to patch Protobuf to disable the feature for the repository.
---
 REPO.bazel | 11 +++++++++++
 1 file changed, 11 insertions(+)
 create mode 100644 REPO.bazel

diff --git a/REPO.bazel b/REPO.bazel
new file mode 100644
index 00000000000..a537ac745fa
--- /dev/null
+++ b/REPO.bazel
@@ -0,0 +1,11 @@
+# This file is read by Bazel 7 and newer, both if Protobuf is the main
+# repository and if it is an external repository.
+repo(
+    features = [
+        # Protobuf cc_* targets do not specify all dependencies from which they
+        # include headers. This causes builds of downstream projects with
+        # --feature=layering_check to fail, which can be avoided by disabling
+        # the feature for the entire repository.
+        "-layering_check",
+    ],
+)
